### PR TITLE
ISPN-11304 Add anchored keys documentation in developer guide

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-11304

### CODE REVIEW

File: anchored-keys/src/main/java/org/infinispan/anchored/configuration/AnchoredKeysConfigurationBuilder.java [hunk 1]
```diff
@@ -42,12 +50,46 @@ public void validate() {
...
+               "Anchored keys only support partition handling mode ALLOW_READ_WRITES, please remove the when-split " +
+               "attribute from your configuration");
+      }
+      rootBuilder.clustering().partitionHandling().whenSplit(PartitionHandling.ALLOW_READ_WRITES);
+
+      Attribute<EntryMergePolicy> mergePolicyAttribute =
+            rootBuilder.clustering().partitionHandling().attributes()
+                       .attribute(PartitionHandlingConfiguration.MERGE_POLICY);
+      if (mergePolicyAttribute.get() != MergePolicy.PREFERRED_NON_NULL && mergePolicyAttribute.isModified()) {
+         throw new CacheConfigurationException(
```
Comment (ryanemerson) [inline anchored-keys/src/main/java/org/infinispan/anchored/configuration/AnchoredKeysConfigurationBuilder.java:88]: These should be i18n logs.

File: documentation/src/main/asciidoc/topics/anchored_keys.adoc [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+The per-key overhead of the ConcurrentHashMap-based on-heap cache,
+assuming a 64-bit JVM with compressed OOPS, would be around 92 bytes:
+32 bytes for `ConcurrentHashMap.Node`, 32 bytes for `MetadataImmortalCacheEntry`,
+24 bytes for `RemoteMetadata`, and 4 bytes in the `ConcurrentHashMap.table` array.
+
+
+=== State transfer
+State transfer does not transfer the actual values, but it still needs
+to transfer all the keys and the anchor owner information.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/anchored_keys.adoc:168]: Maybe: "State transfer does not transfer values, only keys and anchor owner information.

File: documentation/src/main/asciidoc/topics/anchored_keys.adoc [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+Assuming that the values are much bigger compared to the keys,
+the anchor cache's state transfer should also be much faster
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/anchored_keys.adoc:171]:
```suggestion
state transfer for an anchored cache should also be much faster
```

**Comment by danberindei** [comment]
Updated

**Comment by tristantarrant** [comment]
Merged thanks

**Pull request merged**